benchmark/factory/arm-kinova.hpp

```diff
@@ -0,0 +1,91 @@
+  typedef typename crocoddyl::FramePlacementTpl<Scalar> FramePlacement;
+  typedef typename crocoddyl::CostModelAbstractTpl<Scalar> CostModelAbstract;
+  typedef typename crocoddyl::CostModelFramePlacementTpl<Scalar> CostModelFramePlacement;
+  typedef typename crocoddyl::CostModelStateTpl<Scalar> CostModelState;
+  typedef typename crocoddyl::CostModelControlTpl<Scalar> CostModelControl;
+  typedef typename crocoddyl::CostModelSumTpl<Scalar> CostModelSum;
+  typedef typename crocoddyl::ActuationModelFullTpl<Scalar> ActuationModelFull;
+  typedef typename crocoddyl::DifferentialActionModelFreeFwdDynamicsTpl<Scalar> DifferentialActionModelFreeFwdDynamics;
+  typedef typename crocoddyl::IntegratedActionModelEulerTpl<Scalar> IntegratedActionModelEuler;
+
+  // because urdf is not supported with all scalar types.
+  pinocchio::ModelTpl<double> modeld;
+  pinocchio::urdf::buildModel(EXAMPLE_ROBOT_DATA_MODEL_DIR "/kinova_description/robots/kinova.urdf", modeld);
+
+  pinocchio::ModelTpl<Scalar> model(modeld.cast<Scalar>());
+
+  boost::shared_ptr<crocoddyl::StateMultibodyTpl<Scalar> > state =
+      boost::make_shared<crocoddyl::StateMultibodyTpl<Scalar> >(
+          boost::make_shared<pinocchio::ModelTpl<Scalar> >(model));
+
+  FramePlacement Mref(model.getFrameId("j2s6s200_end_effector"),
+                      pinocchio::SE3Tpl<Scalar>(Matrix3s::Identity(), Vector3s(Scalar(0), Scalar(0), Scalar(.4))));
+  boost::shared_ptr<CostModelAbstract> goalTrackingCost = boost::make_shared<CostModelFramePlacement>(state, Mref);
+  boost::shared_ptr<CostModelAbstract> xRegCost = boost::make_shared<CostModelState>(state);
+  boost::shared_ptr<CostModelAbstract> uRegCost = boost::make_shared<CostModelControl>(state);
+
+  // Create a cost model per the running and terminal action model.
+  boost::shared_ptr<CostModelSum> runningCostModel = boost::make_shared<CostModelSum>(state);
+  boost::shared_ptr<CostModelSum> terminalCostModel = boost::make_shared<CostModelSum>(state);
+
+  // Then let's added the running and terminal cost functions
+  runningCostModel->addCost("gripperPose", goalTrackingCost, Scalar(1));
+  runningCostModel->addCost("xReg", xRegCost, Scalar(1e-4));
+  runningCostModel->addCost("uReg", uRegCost, Scalar(1e-4));
+  terminalCostModel->addCost("gripperPose", goalTrackingCost, Scalar(1));
+
+  // We define an actuation model
+  boost::shared_ptr<ActuationModelFull> actuation = boost::make_shared<ActuationModelFull>(state);
+
+  // Next, we need to create an action model for running and terminal knots. The
+  // forward dynamics (computed using ABA) are implemented
+  // inside DifferentialActionModelFullyActuated.
+  boost::shared_ptr<DifferentialActionModelFreeFwdDynamics> runningDAM =
+      boost::make_shared<DifferentialActionModelFreeFwdDynamics>(state, actuation, runningCostModel);
+
+  // VectorXs armature(state->get_nq());
+  // armature << 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.;
+  // runningDAM->set_armature(armature);
```

Comment:
You might want to benchmark the case with armature as well.
As you can see, it will run another part of the code defined in the action model.

Reply:
We're moving armature would move to pinocchio, so we don't have to maintain it here.

Reply:
Sounds good!